copy/copy.go

```diff
@@ -121,12 +153,24 @@ type Options struct {
 	Progress         chan types.ProgressProperties // Reported to when ProgressInterval has arrived for a single artifact+offset.
 	// manifest MIME type of image set by user. "" is default and means use the autodetection to the the manifest MIME type
 	ForceManifestMIMEType string
+	ImageListSelection    ImageListSelection // set to either CopySystemImage (the default), CopyAllImages, or CopySpecificImages to control which instances we copy when the source reference is a list; ignored if the source reference is not a list
+	Instances             []digest.Digest    // if ImageListSelection is CopySpecificImages, copy only these instances and the list itself
+}
+
+// validateImageListSelection returns an error if the passed-in value is not one that we recognize as a valid ImageListSelection value
+func validateImageListSelection(selection ImageListSelection) error {
+	switch selection {
+	default:
+		return errors.Errorf("Invalid value for options.ImageListSelection: %d", selection)
+	case CopySystemImage, CopyAllImages, CopySpecificImages:
+	}
+	return nil
```

Comment:
The return in the outer scope here isn't needed. The go compiler unwraps the branches and won't complain if all have a return path. So we could rewrite the function body into:
```golang
	switch selection {
	case CopySystemImage, CopyAllImages, CopySpecificImages:
		return nil
	default:
		return errors.Errorf("Invalid value for options.ImageListSelection: %d", selection)
	}
```

My brain-parser exited here since its used to the default being the last case.

Reply:
Reordered to `return nil` from the known-values cases, and to put the `default:` clause last.